lib/queue.js

```diff
@@ -858,6 +858,10 @@ Queue.prototype.isPaused = function(isLocal) {
 };
 
 Queue.prototype.run = function(concurrency) {
+  // Check if `concurrency--` will be NaN, which is falsy, so jobs never start processing
+  if (_.isNaN(concurrency - 1)) {
```

Comment:
I think it is better to use the more straightforward isFinite: https://lodash.com/docs/4.17.15#isFinite